engine/core/src/main/java/org/datacleaner/reference/SimpleDictionary.java

```diff
@@ -21,51 +21,147 @@
 
 import java.io.IOException;
 import java.io.ObjectInputStream;
+import java.io.ObjectInputStream.GetField;
+import java.io.Serializable;
+import java.lang.reflect.Field;
 import java.util.Collection;
-import java.util.List;
+import java.util.Iterator;
+import java.util.Objects;
+import java.util.Set;
 
+import org.datacleaner.configuration.DataCleanerConfiguration;
 import org.datacleaner.util.ReadObjectBuilder;
+import org.datacleaner.util.ReadObjectBuilder.Adaptor;
+
+import com.google.common.collect.Iterators;
+import com.google.common.collect.Sets;
 
 /**
- * The simplest possible Dictionary implementation. Based on an in-memory set of
- * values.
- * 
- * 
+ * The simplest possible Dictionary implementation. Based on an in-memory
+ * {@link Set} of values.
  */
 public final class SimpleDictionary extends AbstractReferenceData implements Dictionary {
 
-	private static final long serialVersionUID = 1L;
+    private static final long serialVersionUID = 1L;
+
+    private final Set<String> _valueSet;
+    private final boolean _caseSensitive;
+
+    public SimpleDictionary(String name, String... values) {
+        this(name, true, values);
+    }
+
+    public SimpleDictionary(String name, boolean caseSensitive, String... values) {
+        this(name, createValueSet(values, caseSensitive), caseSensitive);
+    }
+
+    public SimpleDictionary(String name, Collection<String> values) {
+        this(name, values, false);
+    }
+
+    public SimpleDictionary(String name, Collection<String> values, boolean caseSensitive) {
+        super(name);
+        if (caseSensitive) {
+            _valueSet = Sets.newHashSet(values);
+        } else {
+            _valueSet = createValueSet(values.iterator(), caseSensitive);
+        }
+        _caseSensitive = caseSensitive;
+    }
+
+    private static Set<String> createValueSet(Object[] array, boolean caseSensitive) {
+        return createValueSet(Iterators.forArray(array), caseSensitive);
+    }
+
+    private static Set<String> createValueSet(Iterator<?> iterator, boolean caseSensitive) {
+        final Set<String> valueSet = Sets.newHashSet();
+        while (iterator.hasNext()) {
+            final Object value = iterator.next();
+            if (value != null) {
+                if (caseSensitive) {
+                    valueSet.add(value.toString());
+                } else {
+                    valueSet.add(value.toString().toLowerCase());
+                }
+            }
+        }
+        return valueSet;
+    }
+
+    private void readObject(ObjectInputStream stream) throws IOException, ClassNotFoundException {
+        Adaptor adaptor = new Adaptor() {
```

Comment:
final (and the whole variable seems pretty useless, but we probably wont agree on that :))